spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/r2dbc/R2dbcProperties.java

```diff
@@ -138,6 +141,21 @@ public String determineUniqueName() {
 		 */
 		private Duration maxIdleTime = Duration.ofMinutes(30);
 
+		/**
+		 * Max lifetime.
+		 */
+		private Duration maxLifeTime = Duration.ZERO;
```

Comment:
Please use the relevant `of` with `0` instead so that we can extract the proper unit.